pkg/solidity-utils/contracts/openzeppelin/AccessControl.sol

```diff
@@ -45,14 +45,16 @@ abstract contract AccessControl {
     using EnumerableSet for EnumerableSet.AddressSet;
 
     struct RoleData {
-        EnumerableSet.AddressSet members;
+        EnumerableSet.AddressSet globalMembers;
+        mapping(address => EnumerableSet.AddressSet) membersByContract;
         bytes32 adminRole;
     }
 
     mapping(bytes32 => RoleData) private _roles;
 
     bytes32 public constant DEFAULT_ADMIN_ROLE = 0x00;
-
+    address public constant GLOBAL_ROLE_ADMIN = address(0);
```

Comment:
Yes, kind of.

When you check for has role:

```
function hasRole(bytes32 role, address account, address where) public view virtual returns (bool) {
        return _roles[role].globalMembers.contains(account) || 
            _roles[role].membersByContract[where].contains(account);    
}
```
If it has the global role, it will ignore the `where` address, and in case it does not have the global role, you should not set address(0) as a `where` so it will return false. It's a way to make more evident you are checking for a global role without another method.

Actually, just realized we should require not giving a non-global role to address(0) just in case. Will make a commit

Reply:
Ah that's a good realization, yes. I think it is another piece of evidence pointing towards #985 being desirable (particularly, hvaing the notion of adminship be more explicit).